specification/Metadata/README.md

```diff
@@ -0,0 +1,1130 @@
+
+Base64 is easy to transmit as it only uses ASCII characters, but is not very efficient for storage. If large blobs are needed, it is highly recommended to use the binary metadata encoding instead.
+
+```json
+{
+  "classes": {
+    "basicTypes": {
+      "properties": {
+        "blobProperty": {
+          "type": "BLOB"
+        }
+      }
+    }
+  },
+  "instanceTables": {
+    "basicTypesTable": {
+      "class": "basicTypes",
+      "count": 3,
+      "properties": {
+        "blobProperty": {
+          "values": [
+            "TG9va2luZyBmb3Igc2VjcmV0cw==",
+            "WW91IGRlY29kZWQgdGhlIG1lc3NhZ2U=",
+            "QnV0IG5vbmUgY291bGQgYmUgZm91bmQ="
+          ]
+        }
+      }
+    }
+  }
+}
+```
+
+#### Fixed-length Strings and Blobs
+
+In some cases, strings have a known, fixed length. For example, an application may require dates to be entered in `YYYYMMDD` format, which will fit in exactly 8 bytes. For this,
+the specification provides a `stringByteLength` property. In the JSON encoding, this byte length refers to the length of the string when UTF-8 encoded, regardless of how it is deplayed.
+
+`blobByteLength` is similar, but used for binary data. The length is measured as the _decoded_ binary length, not the length of the Base64 string.
+
+```json
+{
+  "classes": {
+    "basicTypes": {
+      "properties": {
+        "date": {
+          "name": "Date (YYYYMMDD)",
+          "type": "STRING",
+          "stringByteLength": 4
+        },
+        "fixedBlobProperty": {
+          "type": "BLOB",
+          "blobByteLength": 4
+        }
+      }
+    }
+  },
+  "instanceTables": {
+    "basicTypesTable": {
+      "class": "basicTypes",
+      "count": 3,
+      "properties": {
+        "blobProperty": {
+          "values": [
+            "AAECAw=="
+            "BAUGBw=="
+            "CAkKCw=="
+          ]
+        }
+      }
+    }
+  }
+}
+```
+
+#### Optional and Default Values
+
+In JSON, optional values can be expressed by using `null` in place of the value. This is true for each data type except for arrays. For arrays, use `[]` when either an array has 0 components or the entire array is undefined (this specification does not distinguish between the two). However, to mark a single component as missing, `null` is still used.
+
+
+```jsonc
+{
+  "classes": {
+    "sensor": {
+      "name": "Temperature Sensor",
+      "properties": {
+        "currentValue": {
+          "name": "Current temperature",
+          "description": "A basic example of using `null` for unknown values",
+          "type": "FLOAT32",
+          "optional": true
+        },
+        "history": {
+          "name": "Previous 5 sensor readings",
+          "description": "This example demonstrates the use of `[]` for null arrays as well as how `null` can be used for individual array components.",
+          "type": "ARRAY",
+          "componentType": "STRING",
+          "optional": true,
```

Comment:
I'd rather see `[null, null, null, null, null]` in this case. The only special case is that `[]` is used instead of `null` to indicate the absence of an array.